integration/integration_test.go

```diff
@@ -145,10 +145,20 @@ func TestHttpIngress(t *testing.T) {
 					assert.Equal(t, []byte("true"), resp.bodyBytes)
 				}),
 				httpCall(rd, http.MethodGet, "/error", nil, func(t testing.TB, resp *httpResponse) {
+					assert.Equal(t, 500, resp.status)
 					assert.Equal(t, []string{"text/plain; charset=utf-8"}, resp.headers["Content-Type"])
 					assert.Equal(t, []byte("Error from FTL"), resp.bodyBytes)
-					assert.Equal(t, 500, resp.status)
 				}),
+				httpCall(rd, http.MethodGet, "/array/string", jsonData(t, []string{"hello", "world"}), func(t testing.TB, resp *httpResponse) {
+					assert.Equal(t, 200, resp.status)
+					assert.Equal(t, []string{"application/json; charset=utf-8"}, resp.headers["Content-Type"])
+					assert.Equal(t, jsonData(t, []string{"hello", "world"}), resp.bodyBytes)
+				}),
+				// httpCall(rd, http.MethodPost, "/array/data", jsonData(t, []obj{{"item": "a"}, {"item": "b"}}), func(t testing.TB, resp *httpResponse) {
+				// 	assert.Equal(t, 200, resp.status)
+				// 	assert.Equal(t, []string{"application/json; charset=utf-8"}, resp.headers["Content-Type"])
+				// 	assert.Equal(t, jsonData(t, []obj{{"item": "a"}, {"item": "b"}}), resp.bodyBytes)
+				// }),
```

Comment:
@alecthomas I have this one commented out because it seems like Arrays of DataRefs fail to monomorphize correctly. I took a stab at fixing it in `data.go`, but I think I was causing more harm than good 😂 

Maybe we can take a peek together on this one.

It seems to happen when monomorphising to get the body of the request object.

```go
func getBodyField(dataRef *schema.DataRef, sch *schema.Schema) (*schema.Field, error) {
	data, err := sch.ResolveDataRefMonomorphised(dataRef)
	if err != nil {
		return nil, err
	}
```